File: local_examples/cmds_set/ruby/cmds_set.rb
# EXAMPLE: cmds_set
# HIDE_START
require 'redis'

r = Redis.new
# HIDE_END

# REMOVE_START
def assert_equal(expected, actual)
  raise "Expected #{expected.inspect}, got #{actual.inspect}" unless actual == expected
end
# REMOVE_END

# STEP_START sadd
# REMOVE_START
r.del('myset')
# REMOVE_END
res1 = r.sadd('myset', ['Hello', 'World'])
puts res1 # >>> 2

res2 = r.sadd('myset', ['World'])
puts res2 # >>> 0

res3 = r.smembers('myset')
puts res3.inspect # >>> ["Hello", "World"]
# STEP_END

# REMOVE_START
assert_equal(2, res1)
assert_equal(0, res2)
assert_equal(['Hello', 'World'], res3.sort)
r.del('myset')
# REMOVE_END

# STEP_START smembers
# REMOVE_START
r.del('myset')
# REMOVE_END
res4 = r.sadd('myset', ['Hello', 'World'])
puts res4 # >>> 2

res5 = r.smembers('myset')
puts res5.inspect # >>> ["Hello", "World"]
# STEP_END

# REMOVE_START
assert_equal(2, res4)
assert_equal(['Hello', 'World'], res5.sort)
r.del('myset')
r.close
# REMOVE_END

File: local_examples/cmds_set/ioredis/cmds-set.js
// EXAMPLE: cmds_set

// HIDE_START
import assert from 'node:assert';
import { Redis } from 'ioredis';

const redis = new Redis();
// HIDE_END

// STEP_START sadd
// REMOVE_START
await redis.del('myset');
// REMOVE_END
const res1 = await redis.sadd('myset', 'Hello', 'World');
console.log(res1); // >>> 2

const res2 = await redis.sadd('myset', 'World');
console.log(res2); // >>> 0

const res3 = await redis.smembers('myset');
console.log(res3); // >>> ['Hello', 'World']
// STEP_END

// REMOVE_START
assert.equal(res1, 2);
assert.equal(res2, 0);
assert.deepEqual([...res3].sort(), ['Hello', 'World']);
await redis.del('myset');
// REMOVE_END

// STEP_START smembers
// REMOVE_START
await redis.del('myset');
// REMOVE_END
const res4 = await redis.sadd('myset', 'Hello', 'World');
console.log(res4); // >>> 2

const res5 = await redis.smembers('myset');
console.log(res5); // >>> ['Hello', 'World']
// STEP_END

// REMOVE_START
assert.equal(res4, 2);
assert.deepEqual([...res5].sort(), ['Hello', 'World']);
await redis.del('myset');
// REMOVE_END

// HIDE_START
redis.disconnect();
// HIDE_END
